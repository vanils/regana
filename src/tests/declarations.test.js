
const regana = require('../regana.js');
const path = require('path');

describe('regana', () => {
  describe('#analyse', () => {
    describe('#declarations', () => {
      test('analyse should work with variable declarations', () => {
        expect(regana.analyseFile(path.resolve(__dirname, './mocks/declarations/variable/mock1.js'))).toMatchSnapshot();
        expect(regana.analyseFile(path.resolve(__dirname, './mocks/declarations/variable/mock2.js'))).toMatchSnapshot();
        expect(regana.analyseFile(path.resolve(__dirname, './mocks/declarations/variable/mock3.js'))).toMatchSnapshot();
        expect(regana.analyseFile(path.resolve(__dirname, './mocks/declarations/variable/mock4.js'))).toMatchSnapshot();
        expect(regana.analyseFile(path.resolve(__dirname, './mocks/declarations/variable/mock5.js'))).toMatchSnapshot();
      });
    });
  });
});