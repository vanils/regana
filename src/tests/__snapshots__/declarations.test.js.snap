// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`regana #analyse #declarations analyse should work with variable declarations 1`] = `
"[
  {
    \\"start\\": 0,
    \\"end\\": 14,
    \\"parentScope\\": null,
    \\"file\\": \\"File_0\\",
    \\"segments\\": {
      \\"NumericLiteral_2\\": {
        \\"uses\\": [],
        \\"start\\": 12,
        \\"end\\": 13,
        \\"id\\": \\"NumericLiteral_2\\"
      },
      \\"VariableDeclarator_1\\": {
        \\"uses\\": [
          \\"NumericLiteral_2\\"
        ],
        \\"start\\": 5,
        \\"end\\": 13,
        \\"id\\": \\"VariableDeclarator_1\\"
      }
    },
    \\"exposes\\": {},
    \\"pointers\\": {
      \\"test\\": \\"VariableDeclarator_1\\"
    },
    \\"withSegments\\": []
  }
]"
`;

exports[`regana #analyse #declarations analyse should work with variable declarations 2`] = `
"[
  {
    \\"start\\": 0,
    \\"end\\": 14,
    \\"parentScope\\": null,
    \\"file\\": \\"File_0\\",
    \\"segments\\": {
      \\"NumericLiteral_2\\": {
        \\"uses\\": [],
        \\"start\\": 12,
        \\"end\\": 13,
        \\"id\\": \\"NumericLiteral_2\\"
      },
      \\"VariableDeclarator_1\\": {
        \\"uses\\": [
          \\"NumericLiteral_2\\"
        ],
        \\"start\\": 5,
        \\"end\\": 13,
        \\"id\\": \\"VariableDeclarator_1\\"
      }
    },
    \\"exposes\\": {},
    \\"pointers\\": {
      \\"test\\": \\"VariableDeclarator_1\\"
    },
    \\"withSegments\\": []
  },
  {
    \\"start\\": 0,
    \\"end\\": 14,
    \\"parentScope\\": null,
    \\"file\\": \\"File_3\\",
    \\"segments\\": {
      \\"NumericLiteral_5\\": {
        \\"uses\\": [],
        \\"start\\": 12,
        \\"end\\": 13,
        \\"id\\": \\"NumericLiteral_5\\"
      },
      \\"VariableDeclarator_4\\": {
        \\"uses\\": [
          \\"NumericLiteral_5\\"
        ],
        \\"start\\": 5,
        \\"end\\": 13,
        \\"id\\": \\"VariableDeclarator_4\\"
      }
    },
    \\"exposes\\": {},
    \\"pointers\\": {
      \\"test\\": \\"VariableDeclarator_4\\"
    },
    \\"withSegments\\": []
  }
]"
`;

exports[`regana #analyse #declarations analyse should work with variable declarations 3`] = `
"[
  {
    \\"start\\": 0,
    \\"end\\": 14,
    \\"parentScope\\": null,
    \\"file\\": \\"File_0\\",
    \\"segments\\": {
      \\"NumericLiteral_2\\": {
        \\"uses\\": [],
        \\"start\\": 12,
        \\"end\\": 13,
        \\"id\\": \\"NumericLiteral_2\\"
      },
      \\"VariableDeclarator_1\\": {
        \\"uses\\": [
          \\"NumericLiteral_2\\"
        ],
        \\"start\\": 5,
        \\"end\\": 13,
        \\"id\\": \\"VariableDeclarator_1\\"
      }
    },
    \\"exposes\\": {},
    \\"pointers\\": {
      \\"test\\": \\"VariableDeclarator_1\\"
    },
    \\"withSegments\\": []
  },
  {
    \\"start\\": 0,
    \\"end\\": 14,
    \\"parentScope\\": null,
    \\"file\\": \\"File_3\\",
    \\"segments\\": {
      \\"NumericLiteral_5\\": {
        \\"uses\\": [],
        \\"start\\": 12,
        \\"end\\": 13,
        \\"id\\": \\"NumericLiteral_5\\"
      },
      \\"VariableDeclarator_4\\": {
        \\"uses\\": [
          \\"NumericLiteral_5\\"
        ],
        \\"start\\": 5,
        \\"end\\": 13,
        \\"id\\": \\"VariableDeclarator_4\\"
      }
    },
    \\"exposes\\": {},
    \\"pointers\\": {
      \\"test\\": \\"VariableDeclarator_4\\"
    },
    \\"withSegments\\": []
  },
  {
    \\"start\\": 0,
    \\"end\\": 16,
    \\"parentScope\\": null,
    \\"file\\": \\"File_6\\",
    \\"segments\\": {
      \\"NumericLiteral_8\\": {
        \\"uses\\": [],
        \\"start\\": 14,
        \\"end\\": 15,
        \\"id\\": \\"NumericLiteral_8\\"
      },
      \\"VariableDeclarator_7\\": {
        \\"uses\\": [
          \\"NumericLiteral_8\\"
        ],
        \\"start\\": 7,
        \\"end\\": 15,
        \\"id\\": \\"VariableDeclarator_7\\"
      }
    },
    \\"exposes\\": {},
    \\"pointers\\": {
      \\"test\\": \\"VariableDeclarator_7\\"
    },
    \\"withSegments\\": []
  }
]"
`;

exports[`regana #analyse #declarations analyse should work with variable declarations 4`] = `
"[
  {
    \\"start\\": 0,
    \\"end\\": 14,
    \\"parentScope\\": null,
    \\"file\\": \\"File_0\\",
    \\"segments\\": {
      \\"NumericLiteral_2\\": {
        \\"uses\\": [],
        \\"start\\": 12,
        \\"end\\": 13,
        \\"id\\": \\"NumericLiteral_2\\"
      },
      \\"VariableDeclarator_1\\": {
        \\"uses\\": [
          \\"NumericLiteral_2\\"
        ],
        \\"start\\": 5,
        \\"end\\": 13,
        \\"id\\": \\"VariableDeclarator_1\\"
      }
    },
    \\"exposes\\": {},
    \\"pointers\\": {
      \\"test\\": \\"VariableDeclarator_1\\"
    },
    \\"withSegments\\": []
  },
  {
    \\"start\\": 0,
    \\"end\\": 14,
    \\"parentScope\\": null,
    \\"file\\": \\"File_3\\",
    \\"segments\\": {
      \\"NumericLiteral_5\\": {
        \\"uses\\": [],
        \\"start\\": 12,
        \\"end\\": 13,
        \\"id\\": \\"NumericLiteral_5\\"
      },
      \\"VariableDeclarator_4\\": {
        \\"uses\\": [
          \\"NumericLiteral_5\\"
        ],
        \\"start\\": 5,
        \\"end\\": 13,
        \\"id\\": \\"VariableDeclarator_4\\"
      }
    },
    \\"exposes\\": {},
    \\"pointers\\": {
      \\"test\\": \\"VariableDeclarator_4\\"
    },
    \\"withSegments\\": []
  },
  {
    \\"start\\": 0,
    \\"end\\": 16,
    \\"parentScope\\": null,
    \\"file\\": \\"File_6\\",
    \\"segments\\": {
      \\"NumericLiteral_8\\": {
        \\"uses\\": [],
        \\"start\\": 14,
        \\"end\\": 15,
        \\"id\\": \\"NumericLiteral_8\\"
      },
      \\"VariableDeclarator_7\\": {
        \\"uses\\": [
          \\"NumericLiteral_8\\"
        ],
        \\"start\\": 7,
        \\"end\\": 15,
        \\"id\\": \\"VariableDeclarator_7\\"
      }
    },
    \\"exposes\\": {},
    \\"pointers\\": {
      \\"test\\": \\"VariableDeclarator_7\\"
    },
    \\"withSegments\\": []
  },
  {
    \\"start\\": 0,
    \\"end\\": 29,
    \\"parentScope\\": null,
    \\"file\\": \\"File_9\\",
    \\"segments\\": {
      \\"NumericLiteral_11\\": {
        \\"uses\\": [],
        \\"start\\": 14,
        \\"end\\": 15,
        \\"id\\": \\"NumericLiteral_11\\"
      },
      \\"VariableDeclarator_10\\": {
        \\"uses\\": [
          \\"NumericLiteral_11\\"
        ],
        \\"start\\": 7,
        \\"end\\": 15,
        \\"id\\": \\"VariableDeclarator_10\\"
      },
      \\"NumericLiteral_13\\": {
        \\"uses\\": [],
        \\"start\\": 27,
        \\"end\\": 28,
        \\"id\\": \\"NumericLiteral_13\\"
      },
      \\"VariableDeclarator_12\\": {
        \\"uses\\": [
          \\"NumericLiteral_13\\"
        ],
        \\"start\\": 17,
        \\"end\\": 28,
        \\"id\\": \\"VariableDeclarator_12\\"
      }
    },
    \\"exposes\\": {},
    \\"pointers\\": {
      \\"test\\": \\"VariableDeclarator_10\\",
      \\"another\\": \\"VariableDeclarator_12\\"
    },
    \\"withSegments\\": []
  }
]"
`;

exports[`regana #analyse #declarations analyse should work with variable declarations 5`] = `
"[
  {
    \\"start\\": 0,
    \\"end\\": 14,
    \\"parentScope\\": null,
    \\"file\\": \\"File_0\\",
    \\"segments\\": {
      \\"NumericLiteral_2\\": {
        \\"uses\\": [],
        \\"start\\": 12,
        \\"end\\": 13,
        \\"id\\": \\"NumericLiteral_2\\"
      },
      \\"VariableDeclarator_1\\": {
        \\"uses\\": [
          \\"NumericLiteral_2\\"
        ],
        \\"start\\": 5,
        \\"end\\": 13,
        \\"id\\": \\"VariableDeclarator_1\\"
      }
    },
    \\"exposes\\": {},
    \\"pointers\\": {
      \\"test\\": \\"VariableDeclarator_1\\"
    },
    \\"withSegments\\": []
  },
  {
    \\"start\\": 0,
    \\"end\\": 14,
    \\"parentScope\\": null,
    \\"file\\": \\"File_3\\",
    \\"segments\\": {
      \\"NumericLiteral_5\\": {
        \\"uses\\": [],
        \\"start\\": 12,
        \\"end\\": 13,
        \\"id\\": \\"NumericLiteral_5\\"
      },
      \\"VariableDeclarator_4\\": {
        \\"uses\\": [
          \\"NumericLiteral_5\\"
        ],
        \\"start\\": 5,
        \\"end\\": 13,
        \\"id\\": \\"VariableDeclarator_4\\"
      }
    },
    \\"exposes\\": {},
    \\"pointers\\": {
      \\"test\\": \\"VariableDeclarator_4\\"
    },
    \\"withSegments\\": []
  },
  {
    \\"start\\": 0,
    \\"end\\": 16,
    \\"parentScope\\": null,
    \\"file\\": \\"File_6\\",
    \\"segments\\": {
      \\"NumericLiteral_8\\": {
        \\"uses\\": [],
        \\"start\\": 14,
        \\"end\\": 15,
        \\"id\\": \\"NumericLiteral_8\\"
      },
      \\"VariableDeclarator_7\\": {
        \\"uses\\": [
          \\"NumericLiteral_8\\"
        ],
        \\"start\\": 7,
        \\"end\\": 15,
        \\"id\\": \\"VariableDeclarator_7\\"
      }
    },
    \\"exposes\\": {},
    \\"pointers\\": {
      \\"test\\": \\"VariableDeclarator_7\\"
    },
    \\"withSegments\\": []
  },
  {
    \\"start\\": 0,
    \\"end\\": 29,
    \\"parentScope\\": null,
    \\"file\\": \\"File_9\\",
    \\"segments\\": {
      \\"NumericLiteral_11\\": {
        \\"uses\\": [],
        \\"start\\": 14,
        \\"end\\": 15,
        \\"id\\": \\"NumericLiteral_11\\"
      },
      \\"VariableDeclarator_10\\": {
        \\"uses\\": [
          \\"NumericLiteral_11\\"
        ],
        \\"start\\": 7,
        \\"end\\": 15,
        \\"id\\": \\"VariableDeclarator_10\\"
      },
      \\"NumericLiteral_13\\": {
        \\"uses\\": [],
        \\"start\\": 27,
        \\"end\\": 28,
        \\"id\\": \\"NumericLiteral_13\\"
      },
      \\"VariableDeclarator_12\\": {
        \\"uses\\": [
          \\"NumericLiteral_13\\"
        ],
        \\"start\\": 17,
        \\"end\\": 28,
        \\"id\\": \\"VariableDeclarator_12\\"
      }
    },
    \\"exposes\\": {},
    \\"pointers\\": {
      \\"test\\": \\"VariableDeclarator_10\\",
      \\"another\\": \\"VariableDeclarator_12\\"
    },
    \\"withSegments\\": []
  },
  {
    \\"start\\": 0,
    \\"end\\": 10,
    \\"parentScope\\": null,
    \\"file\\": \\"File_14\\",
    \\"segments\\": {
      \\"VariableDeclarator_15\\": {
        \\"uses\\": [],
        \\"start\\": 5,
        \\"end\\": 9,
        \\"id\\": \\"VariableDeclarator_15\\"
      }
    },
    \\"exposes\\": {},
    \\"pointers\\": {
      \\"test\\": \\"VariableDeclarator_15\\"
    },
    \\"withSegments\\": []
  }
]"
`;
